sopel/tools/calculation.py

```diff
@@ -176,13 +186,11 @@ def pow_complexity(num, exp):
         return exp ** 1.590 * num.bit_length() ** 1.73 / 36864057619.3
 
 
-def guarded_pow(num, exp):
+def guarded_pow(num: Union[int, float], exp: Union[int, float]):
```

Comment:
Nit/:fire:fun:fire: fact: you can just annotate `float` here, because `int` coerces to `float` just fine as far as typing is concerned, which is [explicitly-defined behavior in PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower):

> when an argument is annotated as having type float, an argument of type int is acceptable

(Unless I'm wrong and you checked that and we actually _do_ need the `Union`, in which case this needs a comment explaining why :stuck_out_tongue_winking_eye:)

```suggestion
def guarded_pow(num: float, exp: float):
```

Reply:
> (Unless I'm wrong and you checked that and we actually _do_ need the `Union`, in which case this needs a comment explaining why 😜)

Nah, I just didn't think of the inheritance there. Found several places with an unnecessary `Union` of this kind—all of them, in fact. The file no longer needs `Union` at all.